projects/backend/image.ts

```diff
@@ -1,35 +1,51 @@
 import { createHash } from 'crypto'
-import { ImageSize } from './common'
+import { ImageSize, Image } from './common'
+import Vibrant from 'node-vibrant'
+
 const salt = process.env.IMAGE_SALT
 
-export const getImageSlug = (url: string) => {
+export const getImageFromURL = (url: string): Image => {
     const parsed = new URL(url)
     const path = parsed.pathname
     const host = parsed.hostname
     const hostparts = host.split('.')
     const source = hostparts[0]
-    return source + path
+    return { source, path }
 }
 
 const getSignature = (path: string) => {
-    console.log(`${salt}${path}`)
     return createHash('md5')
         .update(`${salt}/${path}`)
         .digest('hex')
 }
 
-const getSize = (size: ImageSize) => {
-    switch (size) {
-        case 'small':
-            return 500
-        case 'notsmall':
-            return 2000
-    }
+const sizes: { [k in ImageSize]: number } = {
+    phone: 1000,
+    tablet: 2500,
+    sample: 200,
 }
 
-export const getRedirect = (source: string, size: ImageSize, path: string) => {
-    const newPath = `${path}?q=85&w=${getSize(size)}` //&dpr=2&width=500&auto=format`
-    return `https://i.guim.co.uk/img/${source}/${newPath}&s=${getSignature(
-        newPath,
-    )}`
+export const getImageURL = (image: Image, size: ImageSize) => {
+    const newPath = `${image.path}?q=85&w=${sizes[size]}` //&dpr=2&width=500&auto=format`
```

Comment:
If this is really `85` (also for WebP in Android _via_ `auto=format`?) at `dpr=2` (?), then it’s well too good a quality. Even for Mallard. Will mean slow partial and bundle downloads. Dotcom is too hardcore, but for comparison, it’s using WebP `q=25` at `dpr=2`.

Reply:
Thanks 👍